PlayerInfo/PlayerInfo.cpp

```diff
@@ -130,5 +174,217 @@ namespace Plugin {
         }
     }
 
+    uint32_t PlayerInfo::getSinkAtmosCapability (const JsonObject& parameters, JsonObject& response)
+    {   //sample servicemanager response:
+        LOGINFOMETHOD();
+        bool success = true;
+        dsATMOSCapability_t atmosCapability;
+        try
+        {
+            device::AudioOutputPort aPort = device::Host::getInstance().getAudioOutputPort("HDMI0");
+            if (aPort.isConnected()) {
+                aPort.getSinkDeviceAtmosCapability (atmosCapability);
+                response["atmos_capability"] = (int)atmosCapability;
+            }
+            else {
+                LOGERR("getSinkAtmosCapability failure: HDMI0 not connected!\n");
+                success = false;
+            }
+        }
+        catch(const device::Exception& err)
+        {
+            LOG_DEVICE_EXCEPTION1(string("HDMI0"));
+            success = false;
+        }
+        returnResponse(success);
+    }
+
+    uint32_t PlayerInfo::getSoundMode(const JsonObject& parameters, JsonObject& response)
+    {   //sample servicemanager response:{"success":true,"soundMode":"AUTO (Dolby Digital 5.1)"}
+        LOGINFOMETHOD();
+        string audioPort = parameters["audioPort"].String();//empty value will browse all ports
+
+        if (!checkPortName(audioPort))
+            audioPort = "HDMI0";
+
+        string modeString("");
+        device::AudioStereoMode mode = device::AudioStereoMode::kStereo;  //default to stereo
+
+        try
+        {
+            /* Return the sound mode of the audio ouput connected to the specified videoDisplay */
+            /* Check if HDMI is connected - Return (default) Stereo Mode if not connected */
+            if (audioPort.empty())
+            {
+                if (device::Host::getInstance().getVideoOutputPort("HDMI0").isDisplayConnected())
+                {
+                    audioPort = "HDMI0";
+                }
+                else
+                {
+                    /*  * If HDMI is not connected
+                        * Get the SPDIF if it is supported by platform
+                        * If Platform does not have connected ports. Default to HDMI.
+                    */
+                    audioPort = "HDMI0";
```

Comment:
The concept of a plugin is to have magicals in the config (json part of the plugin header see any other plugin) I think the audioPort here would be such a config value. Default could still be "HDMI0" but at least it is easy to change.